pkg/asset/ignition/machine/master_ignition_customizations.go

```diff
@@ -45,10 +46,12 @@ func (a *MasterIgnitionCustomizations) Generate(dependencies asset.Parents) erro
 	defaultPointerIgnition := pointerIgnitionConfig(installConfig.Config, rootCA.Cert(), "master")
 	savedPointerIgnition := master.Config
 
-	if savedPointerIgnition != defaultPointerIgnition {
+	savedPointerIgnitionJson, err := ignition.Marshal(savedPointerIgnition)
+	defaultPointerIgnitionJson, err := ignition.Marshal(defaultPointerIgnition)
+	if string(savedPointerIgnitionJson) != string(defaultPointerIgnitionJson) {
```

Comment:
Maybe [`reflect.DeepEqual`](https://golang.org/pkg/reflect/#DeepEqual) instead? Not sure if there's much distinction, but DeepEqual is used elsewhere in the installer for e.g. comparing on disk and in-memory assets.

Reply:
When I used the `reflect.DeepEqual`, it returned false consistently in testing. This method from @hardys looks to be more fool-proof, IMHO,

Reply:
Yeah I did consider DeepEqual but decided to just use the exact same format the underlying asset does, e.g the ignition configs [get serialized via ignition.Marshal](https://github.com/openshift/installer/blob/master/pkg/asset/ignition/machine/master.go#L44), and the thing we ultimately care about after loading them is did that json data change, so a string comparison seemed like the simpler and more consistent option.

Reply:
Right, upon reflection, they're equal approaches.